src/overlays/actors/ovl_Obj_Um/z_obj_um.c

```diff
@@ -46,131 +73,1951 @@ const ActorInit Obj_Um_InitVars = {
+    EnHorse* bandit2;
+
+    spawnPoints = Lib_SegmentedToVirtual(path->points);
+    Audio_QueueSeqCmd(0x8000 | NA_BGM_CHASE);
+
+    bandit1 = (EnHorse*)Actor_Spawn(&play->actorCtx, play, ACTOR_EN_HORSE, spawnPoints[0].x, spawnPoints[0].y,
+                                    spawnPoints[0].z, 0, this->dyna.actor.shape.rot.y, 0,
+                                    ENHORSE_PARAMS(ENHORSE_PARAM_BANDIT, ENHORSE_19));
+    this->bandit1 = bandit1;
+
+    bandit1->unk_540 = bandit1->actor.world.pos;
+
+    bandit1->unk_54C = 0xF;
+    bandit1->unk_550 = 10;
+
+    bandit1->unk_554 = this->pathIndex;
+    bandit1->unk_568 = 0.0f;
+    bandit1->unk_56C = 0.0f;
+    bandit1->unk_558 = 0;
+    bandit1->unk_55C = 40;
+    bandit1->unk_560 = 40;
+
+    bandit1->banditPosition = gZeroVec3f;
+    bandit1->unk_57C = gZeroVec3f;
+
+    bandit1->unk_588 = this->dyna.actor.shape.rot.y;
+    bandit1->curRaceWaypoint = 1;
+
+    bandit2 = (EnHorse*)Actor_Spawn(&play->actorCtx, play, ACTOR_EN_HORSE, spawnPoints[1].x, spawnPoints[1].y,
+                                    spawnPoints[1].z, 0, this->dyna.actor.shape.rot.y, 0,
+                                    ENHORSE_PARAMS(ENHORSE_PARAM_BANDIT, ENHORSE_20));
+    this->bandit2 = bandit2;
+
+    bandit2->unk_540 = bandit2->actor.world.pos;
+
+    bandit2->unk_54C = 0xF;
+    bandit2->unk_550 = 8;
+
+    bandit2->unk_554 = this->pathIndex;
+    bandit2->unk_568 = 0.0f;
+    bandit2->unk_56C = 0.0f;
+    bandit2->unk_55C = 40;
+    bandit2->unk_560 = 40;
+
+    bandit2->unk_57C = gZeroVec3f;
+
+    bandit2->unk_588 = this->dyna.actor.shape.rot.y;
+    bandit2->curRaceWaypoint = 2;
+
+    this->potsLife[0] = 4;
+    this->potsLife[1] = 4;
+    this->potsLife[2] = 4;
+
+    return 0;
+}
+
+typedef struct {
+    /* 0x00 */ s32 unk_00;
+    /* 0x04 */ s32 unk_04;
+    /* 0x08 */ s32 unk_08;
+    /* 0x0C */ f32 unk_0C;
+    /* 0x10 */ s32 unk_10;
+} struct_80B7C254; // size = 0x14
+
+struct_80B7C254 D_80B7C164[] = {
+    { 2, 0, 0, 1.0f, 40 },   { 4, 1, 0, 1.0f, 40 },  { 3, 2, 0, 1.0f, 40 },   { 3, 4, 0, 1.0f, 40 },
+    { 5, 3, 1, -1.0f, 30 },  { 7, 3, 1, 1.0f, 30 },  { 0, 13, 1, -1.0f, 60 }, { 1, 14, 1, 1.0f, 60 },
+    { 13, 10, 0, 1.0f, 40 }, { 14, 8, 0, 1.0f, 40 }, { 8, 5, 0, 1.0f, 30 },   { 10, 7, 0, 1.0f, 30 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Um/func_80B7BABC.s")
+// BanditAttack?
+s32 func_80B781DC(ObjUm* this, EnHorse* bandit1, EnHorse* bandit2, PlayState* play) {
+    s32 temp_v0;
+    s32 phi_s3 = -1;
+    s32 phi_s4 = 0;
+    s32 phi_s2 = 0;
+    f32 phi_f20 = 0.0f;
+    s32 i;
+    s32 mask;
+
+    for (i = 0; i < ARRAY_COUNT(D_80B7C164); i++) {
+        if (bandit1->unk_550 == D_80B7C164[i].unk_00) {
+            if (bandit2->unk_550 != D_80B7C164[i].unk_04) {
+                if (D_80B7C164[i].unk_00 != 3) {
+                    if (D_80B7C164[i].unk_04 != 3 ||
+                        ((mask = Player_GetMask(play)), PLAYER_MASK_CIRCUS_LEADER != mask)) {
```

Comment:
Hmm, this looks suspicious. I wonder if this is the cause of the fake match here.

Reply:
Writing it the logic way swaps two registers
https://decomp.me/scratch/fycaI